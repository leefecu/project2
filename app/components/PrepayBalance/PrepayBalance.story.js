import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import PrepayBalance from './PrepayBalance';
import Styles from '../../Stories.styles';

storiesOf('Component | PrepayBalance', module).add('Default', () => {
  return (
    <View style={Styles.containerWithBG}>
      <View style={Styles.row}>
        <Text style={Styles.title}>Default</Text>
        <PrepayBalance balance={30.98} loading={false} />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>Error</Text>
        <PrepayBalance balance={-1} loading={false} />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>Loading</Text>
        <PrepayBalance balance={30.98} loading={true} />
      </View>
    </View>
  );
});
