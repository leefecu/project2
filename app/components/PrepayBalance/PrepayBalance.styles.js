import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: Colors.azureBlue,
    flexDirection: 'row',
    justifyContent: 'center',
    paddingVertical: Metrics.smallPadding,
    width: Metrics.width,
  },
  leftColumn: {
    borderRightColor: Colors.separatorColor,
    borderRightWidth: 1,
    flex: 3,
    paddingVertical: Metrics.smallestPadding,
    paddingHorizontal: Metrics.smallPadding,
  },
  rightColumn: {
    flex: 7,
    paddingHorizontal: Metrics.smallPadding,
  },
  label: {
    color: Colors.white,
    fontSize: Metrics.Fonts.smallest,
    textAlign: 'center',
    fontFamily: 'CircularPro-Book',
  },
  balance: {
    color: Colors.white,
    fontSize: Metrics.Fonts.small,
    fontFamily: 'CircularPro-Bold',
  },
  loadingIcon: {
    alignSelf: 'flex-start',
    flex: 1,
  },
});
