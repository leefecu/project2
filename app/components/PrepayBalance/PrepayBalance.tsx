import React from 'react';
import { ActivityIndicator, Text, View } from 'react-native';

import { testProps } from '../../helpers/general';

import Styles from './PrepayBalance.styles';

import Colors from '../../themes/Colors';

interface Props {
  balance: number;
  loading: boolean;
}

const PrepayBalance = ({ balance, loading }: Props) => {
  return (
    <View style={Styles.container}>
      <View style={Styles.leftColumn}>
        <Text style={Styles.label}>Prepay balance</Text>
      </View>
      <View style={Styles.rightColumn}>
        {loading ? (
          <ActivityIndicator style={Styles.loadingIcon} size="small" color={Colors.white} />
        ) : (
          <Text style={Styles.balance} {...testProps('Prepay Balance')}>
            {balance !== -1 ? `$${balance.toFixed(2)}` : 'N/A'}
          </Text>
        )}
      </View>
    </View>
  );
};

PrepayBalance.defaultProps = {
  loading: false,
};

export default PrepayBalance;
