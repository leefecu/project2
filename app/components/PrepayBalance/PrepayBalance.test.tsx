/**
 * @jest-environment jsdom
 */
import React from 'react';
import { create } from 'react-test-renderer';

import PrepayBalance from './PrepayBalance';

describe('PrepayBalance component', () => {
  test('render with balance', () => {
    const button = create(<PrepayBalance balance={30.98} loading={false} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('render error state', () => {
    const button = create(<PrepayBalance balance={-1} loading={false} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('render loading status', () => {
    const button = create(<PrepayBalance balance={30.98} loading={true} />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
