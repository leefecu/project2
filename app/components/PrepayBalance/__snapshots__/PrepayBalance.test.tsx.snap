// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrepayBalance component render error state 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d8bc6",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingVertical": 12,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "#2F97C9",
        "borderRightWidth": 1,
        "flex": 3,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "CircularPro-Book",
          "fontSize": 13,
          "textAlign": "center",
        }
      }
    >
      Prepay balance
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 7,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      accessibilityLabel="Prepay Balance"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 17,
        }
      }
      testID="Prepay Balance"
    >
      N/A
    </Text>
  </View>
</View>
`;

exports[`PrepayBalance component render loading status 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d8bc6",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingVertical": 12,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "#2F97C9",
        "borderRightWidth": 1,
        "flex": 3,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "CircularPro-Book",
          "fontSize": 13,
          "textAlign": "center",
        }
      }
    >
      Prepay balance
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 7,
        "paddingHorizontal": 12,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "alignSelf": "flex-start",
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;

exports[`PrepayBalance component render with balance 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d8bc6",
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingVertical": 12,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "#2F97C9",
        "borderRightWidth": 1,
        "flex": 3,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "CircularPro-Book",
          "fontSize": 13,
          "textAlign": "center",
        }
      }
    >
      Prepay balance
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 7,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      accessibilityLabel="Prepay Balance"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "CircularPro-Bold",
          "fontSize": 17,
        }
      }
      testID="Prepay Balance"
    >
      $30.98
    </Text>
  </View>
</View>
`;
